src/prime.py

```diff
@@ -95,11 +108,15 @@ def target():
 
     Spaces are variable digits
     """
-    filename = default_input('Template filename: ', '../templates/demo.txt')
-    nthreads = int(default_input('Number of threads: ', '1'))
+    parser = create_arg_parser()
+    args = parser.parse_args(sys.argv[1:])
+    filename = args.template
+    nthreads = args.concurrency
 
     template, height, width = parse_template(filename)
 
+    if not os.path.exists('../tmp/'):
+        os.mkdir('../tmp') # TODO: find a better way, ot at least make this directory more obvious to user
```

Comment:
1. should the program quit once one prime is found?
2. or is it worth it to search for all (or ~3) of them so the user can look at them and pick a favorite?